doc/containers.md

```diff
@@ -235,6 +235,33 @@ lxc config device add <container> <device-name> nic nictype=sriov parent=<sriov-
 To tell LXD to use a specific unused VF add the `host_name` property and pass
 it the name of the enabled VF.
 
+### Type: infiniband
+LXD supports two different kind of network types for infiniband devices:
+
+ - `physical`: Straight physical device passthrough from the host. The targeted device will vanish from the host and appear in the container.
+ - `sriov`: Passes a virtual function of an SR-IOV enabled physical network device into the container.
+
+Different network interface types have different additional properties, the current list is:
+
+Key                     | Type      | Default           | Required  | Used by         | API extension | Description
+:--                     | :--       | :--               | :--       | :--             | :--           | :--
+nictype                 | string    | -                 | yes       | all             | infiniband    | The device type, one of "physical", or "sriov"
+name                    | string    | kernel assigned   | no        | all             | infiniband    | The name of the interface inside the container
+hwaddr                  | string    | randomly assigned | no        | all             | infiniband    | The MAC address of the new interface
+mtu                     | integer   | parent MTU        | no        | all             | infiniband    | The MTU of the new interface
+parent                  | string    | -                 | yes       | physical, sriov | infiniband    | The name of the host device or bridge
+
```

Comment:
Probably worth a short paragraph on the fact that SR-IOV on infiniband isn't dynamic so the user needs to pre-configure the number of functions and kernel in their driver/tools before they can make use of them in LXD.

Reply:
addressed